plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -176,14 +177,13 @@ private List<Slice> rewriteFile(Path sourcePath, FileDeletion deletion)
         try {
             Path rootTablePath = new Path(rootTableLocation);
             String sourceRelativePath = rootTablePath.toUri().relativize(sourcePath.toUri()).toString();
-            FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsContext(session.getIdentity()), rootTablePath);
 
             Path targetPath = new Path(sourcePath.getParent(), session.getQueryId() + "_" + randomUUID());
             String targetRelativePath = rootTablePath.toUri().relativize(targetPath.toUri()).toString();
-            FileWriter fileWriter = createParquetFileWriter(fileSystem, targetPath, dataColumns);
+            FileWriter fileWriter = createParquetFileWriter(targetPath.toString(), dataColumns);
 
             DeltaLakeWriter writer = new DeltaLakeWriter(
-                    fileSystem,
+                    hdfsEnvironment.getFileSystem(new HdfsContext(session.getIdentity()), rootTablePath),
```

Comment:
Do we still need `hdfsEnvironment`?

Reply:
It is not that simple as it requires using `trino-parquet` MetadataReader instead of hive's one but I will give it a try